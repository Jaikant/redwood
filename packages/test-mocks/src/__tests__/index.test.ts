import MockProject from '../index'

jest.mock('mock-fs', () => () => {})

describe('test-mocks', () => {
  const mockrw = new MockProject()

  it('provides a default mock filesystem', () => {
    expect(mockrw.paths).toMatchSnapshot()
  })

  it('allows the filesystem to be mutated', () => {
    mockrw.update((paths) => {
      paths['redwood.toml'] = ''
      return paths
    })
    expect(mockrw.paths).toMatchSnapshot()
  })

  it('allows the filesystem to be updated with merge', () => {
    mockrw.merge(() => {
      return {
        'another-file.text': 'hello',
        web: {
          src: {
            pages: {
              'HelloPage/HelloPage.js': '',
            },
          },
        },
      }
    })
    expect(mockrw.paths).toMatchSnapshot()
  })
})
