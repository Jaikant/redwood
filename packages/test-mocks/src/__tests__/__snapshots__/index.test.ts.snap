// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-mocks allows the filesystem to be mutated 1`] = `
Object {
  "api": Object {
    "src": Object {
      "functions": Object {
        "graphql.js": Object {},
      },
    },
  },
  "redwood.toml": "",
  "web": Object {
    "src": Object {},
  },
}
`;

exports[`test-mocks allows the filesystem to be updated with merge 1`] = `
Object {
  "another-file.text": "hello",
  "api": Object {
    "src": Object {
      "functions": Object {
        "graphql.js": Object {},
      },
    },
  },
  "redwood.toml": "",
  "web": Object {
    "src": Object {
      "pages": Object {
        "HelloPage/HelloPage.js": "",
      },
    },
  },
}
`;

exports[`test-mocks provides a default mock filesystem 1`] = `
Object {
  "api": Object {
    "src": Object {
      "functions": Object {
        "graphql.js": Object {},
      },
    },
  },
  "redwood.toml": "
  [web]
    port = 8910
    apiProxyPath = \\"/.netlify/functions\\"
  [api]
    port = 8911
  [browser]
    open = true
  ",
  "web": Object {
    "src": Object {},
  },
}
`;
